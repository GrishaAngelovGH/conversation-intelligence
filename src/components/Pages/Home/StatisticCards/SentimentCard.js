import { Chart as ChartJS, ArcElement, Tooltip, Legend } from 'chart.js'
import { Doughnut } from 'react-chartjs-2'

import StatisticCard from './StatisticCard'

ChartJS.register(ArcElement, Tooltip, Legend)

export const data = {
  labels: ['Positive', 'Neutral', 'Negative'],
  datasets: [
    {
      label: 'Sentiment',
      data: [75, 25, 0],
      backgroundColor: [
        'rgba(75, 192, 192, 0.2)',
        'rgba(255, 206, 86, 0.2)',
        'rgba(255, 99, 132, 0.2)'
      ],
      borderColor: [
        'rgba(75, 192, 192, 1)',
        'rgba(255, 206, 86, 1)',
        'rgba(255, 99, 132, 1)'
      ],
      borderWidth: 1
    }
  ]
}

const SentimentCard = () => {
  const chart = (
    <Doughnut
      data={data}
      options={{
        responsive: true,
        maintainAspectRatio: false
      }}
    />
  )

  const description = (
    <div>
      <h4>35 sales calls have noticeably higher positive sentiment than average</h4>
      <p className='text-muted'>Negative sentiment might indicate that customers are expressing pain points, which could present coaching opportunities.</p>
    </div>
  )

  return (<StatisticCard chart={chart} description={description} />)
}

export default SentimentCard