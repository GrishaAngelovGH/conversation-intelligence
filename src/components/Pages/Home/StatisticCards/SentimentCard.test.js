import { render } from '@testing-library/react'

import SentimentCard from './SentimentCard'

test('should render component', () => {
  const container = render(<SentimentCard />)

  expect(container).toMatchSnapshot()
})