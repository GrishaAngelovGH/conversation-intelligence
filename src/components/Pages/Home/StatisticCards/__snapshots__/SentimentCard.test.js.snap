// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row align-items-center"
      >
        <div
          class="col-md-6"
        >
          <canvas
            height="0"
            role="img"
            style="display: block; box-sizing: border-box; height: 0px; width: 0px;"
            width="0"
          />
        </div>
        <div
          class="col-md-6"
        >
          <div>
            <h4>
              35 sales calls have noticeably higher positive sentiment than average
            </h4>
            <p
              class="text-muted"
            >
              Negative sentiment might indicate that customers are expressing pain points, which could present coaching opportunities.
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row align-items-center"
    >
      <div
        class="col-md-6"
      >
        <canvas
          height="0"
          role="img"
          style="display: block; box-sizing: border-box; height: 0px; width: 0px;"
          width="0"
        />
      </div>
      <div
        class="col-md-6"
      >
        <div>
          <h4>
            35 sales calls have noticeably higher positive sentiment than average
          </h4>
          <p
            class="text-muted"
          >
            Negative sentiment might indicate that customers are expressing pain points, which could present coaching opportunities.
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
